bubblegum/program/tests/simple.rs

```diff
@@ -21,76 +24,83 @@ const MAX_BUF_SIZE: usize = 64;
 
 // Creates a `BubblegumTestContext`, a `Tree` with default arguments, and also mints an NFT
 // with the default `LeafArgs`.
-async fn context_tree_and_leaf() -> Result<(
+async fn context_tree_and_leaves() -> Result<(
     BubblegumTestContext,
     Tree<MAX_DEPTH, MAX_BUF_SIZE>,
-    LeafArgs,
+    Vec<LeafArgs>,
 )> {
     let context = BubblegumTestContext::new().await?;
 
-    let (tree, leaf) = context
-        .default_create_and_mint::<MAX_DEPTH, MAX_BUF_SIZE>()
+    let (tree, leaves) = context
+        .default_create_and_mint::<MAX_DEPTH, MAX_BUF_SIZE>(DEFAULT_NUM_MINTS)
         .await?;
 
-    Ok((context, tree, leaf))
+    Ok((context, tree, leaves))
 }
 
 #[tokio::test]
 async fn test_create_tree_and_mint_passes() {
-    context_tree_and_leaf().await.unwrap();
+    let (context, tree, _) = context_tree_and_leaves().await.unwrap();
+
+    let payer = context.payer();
+
+    let cfg = tree.read_tree_config().await.unwrap();
+    assert_eq!(cfg.tree_creator, payer.pubkey());
+    assert_eq!(cfg.tree_delegate, payer.pubkey());
+    assert_eq!(cfg.total_mint_capacity, 1 << MAX_DEPTH);
+    assert_eq!(cfg.num_minted, DEFAULT_NUM_MINTS);
 }
 
 #[tokio::test]
```

Comment:
Shouldn't we still add some `asserts` even in the happy path tests that validate the on-chain data is correct?

For example, what if someone made a code change to bubblegum that prevented the `spl-account-compression` CPI from happening but the bubblegum instruction returned `Ok()`?  To prevent that bug wouldn't we want to get the merkle root from the merkle tree, calculate the merkle root of our updated local proof tree, and assert that they match?

Reply:
Yes that's something I want to add in another PR; for now wanted to bring these changes in since they touch a bunch of stuff that influences how tests are actually written.

Reply:
Ok sounds good

Reply:
ok sounds good